waku/waku_archive/archive.nim

```diff
@@ -64,186 +72,145 @@ method validate*(validator: DefaultMessageValidator, msg: WakuMessage): Validati
 
   ok()
 
-## Archive
-
-type
-  WakuArchive* = ref object
-    driver*: ArchiveDriver  # TODO: Make this field private. Remove asterisk
-    validator: MessageValidator
-    retentionPolicy: RetentionPolicy
-    retPolicyFut: Future[Result[void, string]]  ## retention policy cancelable future
-    retMetricsRepFut: Future[Result[void, string]]  ## metrics reporting cancelable future
-
-proc new*(T: type WakuArchive,
-          driver: ArchiveDriver,
-          retentionPolicy = none(RetentionPolicy)):
-          Result[T, string] =
-
-  let retPolicy = if retentionPolicy.isSome():
-                    retentionPolicy.get()
-                  else:
-                    nil
-
-  let wakuArch = WakuArchive(driver: driver,
-                             validator: DefaultMessageValidator(),
-                             retentionPolicy: retPolicy)
-  return ok(wakuArch)
-
-proc handleMessage*(w: WakuArchive,
+proc handleMessage*(self: WakuArchive,
                     pubsubTopic: PubsubTopic,
                     msg: WakuMessage) {.async.} =
   if msg.ephemeral:
     # Ephemeral message, do not store
     return
 
-  if not w.validator.isNil():
-    let validationRes = w.validator.validate(msg)
-    if validationRes.isErr():
-      waku_archive_errors.inc(labelValues = [validationRes.error])
-      return
+  self.validate(msg).isOkOr:
+    waku_archive_errors.inc(labelValues = [error])
+    return
 
+  let
+    msgDigest = computeDigest(msg)
+    msgHash = computeMessageHash(pubsubTopic, msg)
+
+  trace "handling message",
+    pubsubTopic=pubsubTopic,
+    contentTopic=msg.contentTopic,
+    timestamp=msg.timestamp,
+    digest=toHex(msgDigest.data),
+    messageHash=toHex(msgHash)
+  
   let insertStartTime = getTime().toUnixFloat()
 
-  block:
-    let
-      msgDigest = computeDigest(msg)
-      msgHash = computeMessageHash(pubsubTopic, msg)
-      msgDigestHex = toHex(msgDigest.data)
-      msgHashHex = toHex(msgHash)
-      msgReceivedTime = if msg.timestamp > 0: msg.timestamp
-                        else: getNanosecondTime(getTime().toUnixFloat())
-
-    trace "handling message", pubsubTopic=pubsubTopic, contentTopic=msg.contentTopic, timestamp=msg.timestamp, digest=msgDigestHex, messageHash=msgHashHex
-
-    let putRes = await w.driver.put(pubsubTopic, msg, msgDigest, msgHash, msgReceivedTime)
-    if putRes.isErr():
-      if "duplicate key value violates unique constraint" in putRes.error:
-        trace "failed to insert message", err=putRes.error
-      else:
-        debug "failed to insert message", err=putRes.error
-      waku_archive_errors.inc(labelValues = [insertFailure])
-
+  (await self.driver.put(pubsubTopic, msg, msgDigest, msgHash, msg.timestamp)).isOkOr:
+    waku_archive_errors.inc(labelValues = [insertFailure])
+    
+    # TODO diff log lvl per error value WTF is this???
```

Comment:
haha yes I agree that it looks weird.

Let me bring some reasoning about that:

It is mostly to prevent spamming the logs when multiple nodes are connected to the same database. In that case, the message cannot be inserted but is an expected "insert error" and therefore we reduce its visibility by having the log in `trace` level.

The change was introduced in: https://github.com/waku-org/nwaku/commit/574cdf555adcdfc77291902d413bb8884bc0c925
```suggestion
```

Reply:
Make sense! I will add a comment.